packages/discovery-react-components/src/components/SearchResults/utils/getDocumentTitle.ts

```diff
@@ -0,0 +1,18 @@
+import DiscoveryV2 from 'ibm-watson/discovery/v2';
+import get from 'lodash/get';
+
+export const getDocumentTitle = (
+  resultTitleField: string,
+  result?: DiscoveryV2.QueryResult
+): string => {
+  if (result) {
+    const documentId: string | undefined = get(result, 'document_id');
+    const filename: string | undefined = get(result, 'extracted_metadata.filename');
+    const userSelectedTitle: string | undefined = get(result, resultTitleField);
+    const title: string = userSelectedTitle
+      ? userSelectedTitle
+      : get(result, 'extracted_metadata.title') || filename || documentId;
+    return title;
+  }
+  return 'No Title';
```

Comment:
I don't think we'll ever hit this (why show a title if we have no document?), but I don't like this default. Either return `null` or an empty string (I prefer the former).